src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -107,27 +110,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     // creates arraylist and starts query
-    ArrayList<String> groceryList = new ArrayList<String>();
+    ArrayList<Grocery> groceryList = new ArrayList<>();
     long familyID = (long) userInfoEntity.getProperty(FAMILY_ID);
     Query groceryQuery = new Query(GROCERY)
       .setFilter(new Query.FilterPredicate(FAMILY_ID, Query.FilterOperator.EQUAL, familyID));   
     PreparedQuery familyGrocery= datastore.prepare(groceryQuery);
-    groceryList = checkGroceries(familyGrocery);   
+    groceryList = checkGroceries(familyGrocery, userEmail);   
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(groceryList));
     }
 
-  // returns items from the query that match the users familyID
-  private ArrayList<String> checkGroceries(PreparedQuery familyGrocery) {
-    ArrayList<String> groceryList = new ArrayList<String>();
+  // returns items from the query that match the users familyID in the Grocery object    
+  private ArrayList<Grocery> checkGroceries(PreparedQuery familyGrocery, String userEmail) {
+    boolean match;
+    long timestamp = System.currentTimeMillis();
+    System.out.println(timestamp);
+    ArrayList<Grocery> groceryList = new ArrayList<>();
     for (Entity entity : familyGrocery.asIterable()) {
-        String groceryOutput;
-            String groceryItem = (String) entity.getProperty("grocery");
-            String memberEmail = (String) entity.getProperty("assignEmail");
-            groceryOutput = groceryItem + " assigned to: " + memberEmail;
-            groceryList.add(groceryOutput);
+        String groceryItem = (String) entity.getProperty(GROCERY);
+        String assignEmail = (String) entity.getProperty("assignEmail");
+        long id = entity.getKey().getId();     
+        if(userEmail.equals(assignEmail)) {
+            match = true;
+        } else if(assignEmail.equals(" ")){
+            match = true;
```

Comment:
hmm, why is this true?

Maybe leave this `false`? You can always check for the empty assignment on the client side by checking for emptiness.

Reply:
I have it true here so it knows what to do in the js

Reply:
better version of this comment at https://github.com/googleinterns/step122-2020/pull/24#discussion_r453102589

My suggestion is rather than having `match` represent "empty || assigned to current user", just have it represent "assigned to current user". On the javascript, you can check for an empty value for assignEmail to determine if it's assigned to a user.